library/Fission/Web/Routes.hs

```diff
@@ -41,3 +43,7 @@ type IPFSRoute  = IPFSPrefix :> IPFS.API
 type IPFSPrefix = "ipfs"
 
 type PingRoute = "ping" :> Ping.API
+
+type DomainRoute = "domain" 
+                  :> BasicAuth "existing user" User
+                  :> Domain.API
```

Comment:
![](https://media.giphy.com/media/l3q2zVr6cu95nF6O4/giphy.gif)

Reply:
🤔 We may want to factor out `BasicAuth "existing user" User`. Perhaps:

```haskell
module Fission.Web.Auth
-- ...
type ExistingUser = BasicAuth "existing user" User

----------

type AuthRoute = "auth"
               :> "verify"
               :> Auth.ExistingUser
               :> Auth.Verify.API


type DomainRoute = "domain" 
                  :> Auth.ExistingUser
                  :> Domain.API
```

Thoughts?

Reply:
Big fan. I'll do the same with `"heroku add-on api"` for consistency